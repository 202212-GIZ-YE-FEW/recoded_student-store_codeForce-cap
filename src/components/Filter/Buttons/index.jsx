import Image from "next/image"

export default function Buttons({ filters = [], handleCategoryFilter }) {
  const CategoryFilter = ({ filter, image, alt }) => (
    <button className='ml-4' onClick={() => handleCategoryFilter(filter)}>
      <Image
        className='rounded-lg'
        src={image}
        alt={alt}
        width={150}
        height={150}
      />
    </button>
  )

  return (
    <div className='flex flex-col font-poppins mb-10'>
      <div className='grid grid-cols-2 gap-x-10 text-[7vw] md:text-[5vw] lg:text-[3vw] text-bubble-gum font-bold py-5 text-center lg:text-left ml-[28vw] md:ml-[26vw] lg:ml-0 w-full'>
        <span className='bg-high-lighter bg-[length:45vw_100%] md:bg-[length:32vw_100%] lg:bg-[length:50%_100%] bg-no-repeat md:bg-center lg:bg-left-bottom'>
          <button className='lg:ml-9' onClick={() => handleCategoryFilter("")}>
            Categories
          </button>
        </span>
        <span className='bg-high-lighter bg-[length:50%_100%] bg-[center] bg-no-repeat hidden lg:block text-center'>
          <h2>Price Filter</h2>
        </span>
      </div>
      <div className='flex flex-col lg:flex-row justify-between shadow-2xl rounded-[8vw] lg:rounded-full items-center lg:pr-20 py-10'>
        <span className='grid grid-cols-4 gap-4 md:gap-28 lg:gap-4 pb-10 lg:pb-0 mx-auto pr-4 lg:px-0'>
          {filters.map(({ filter, image, alt }) => (
            <CategoryFilter
              key={filter}
              filter={filter}
              image={image}
              alt={alt}
            />
          ))}
        </span>
        <span className='w-[70vw] md:w-[40vw] lg:w-[400px]'>
          <Image
            src='/Images/Price-Filter.png'
            alt='...'
            width={400}
            height={400}
          />
        </span>
      </div>
    </div>
  )
}
