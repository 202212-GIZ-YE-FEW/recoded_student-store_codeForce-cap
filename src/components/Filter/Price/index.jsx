import { useState } from "react"
import { Range } from "react-range"

export default function PriceFilter() {
  const [tempValues, setTempValues] = useState([270, 770]) // temporary min and max values for changes made by user

  const handleChange = (newValues) => {
    setTempValues(newValues)
  }

  const handleMinChange = (e) => {
    const newMin = +e.target.value.replace(/\D/g, "")
    if (newMin < 0) {
      alert("Sorry, the minimum price is $10")
    } else if (newMin > tempValues[1]) {
      alert("Sorry, the minimum price cannot be greater than the maximum price")
    } else {
      setTempValues([newMin, tempValues[1]])
    }
  }

  const handleMaxChange = (e) => {
    const newMax = +e.target.value.replace(/\D/g, "")
    if (newMax > 1000) {
      alert("Sorry, the maximum price is $1000")
    } else if (newMax < tempValues[0]) {
      alert("Sorry, the maximum price cannot be less than the minimum price")
    } else {
      setTempValues([tempValues[0], newMax])
    }
  }

  return (
    <div className='flex flex-col items-stretch'>
      <div className='flex justify-around text-2xl text-bubble-gum font-poppins font-extrabold ml-5 md:ml-10'>
        <input
          type='text'
          value={`$${tempValues[0]}`}
          onChange={handleMinChange}
          className='w-24 rounded-md px-2 py-1 outline-none focus:outline-[#FF8A57] focus:outline-offset-2'
        />
        <input
          type='text'
          value={`$${tempValues[1]}`}
          onChange={handleMaxChange}
          className='w-24 rounded-md px-2 py-1 outline-none focus:outline-[#FF8A57] focus:outline-offset-2'
        />
      </div>
      <div className='mt-4 mx-4 md:mx-8 w-[90%]'>
        <Range
          step={1}
          min={0}
          max={1000}
          values={tempValues}
          onChange={handleChange}
          renderTrack={({ props, children }) => (
            <div {...props} className='py-1.5 bg-[#FF8A57] rounded-full'>
              {children}
            </div>
          )}
          renderThumb={({ props }) => (
            <div
              {...props}
              className='bg-[#7874F2] p-3 rounded-full outline-none cursor-pointer'
            />
          )}
        />
      </div>
    </div>
  )
}

/* Just leave it i need to deal with it in another time */

// import Slider from "@material-ui/core/Slider"
// import { useState } from "react"

// export default function PriceFilter() {
//   const [tempValues, setTempValues] = useState([270, 770]) // temporary min and max values for changes made by user

//   const handleChange = (event, newValues) => {
//     setTempValues(newValues)
//   }

//   const handleMinChange = (event, newMin) => {
//     if (newMin < 10) {
//       alert("Sorry, the minimum price is $10")
//     } else if (newMin > tempValues[1]) {
//       alert("Sorry, the minimum price cannot be greater than the maximum price")
//     } else {
//       setTempValues([newMin, tempValues[1]])
//     }
//   }

//   const handleMaxChange = (event, newMax) => {
//     if (newMax > 1000) {
//       alert("Sorry, the maximum price is $1000")
//     } else if (newMax < tempValues[0]) {
//       alert("Sorry, the maximum price cannot be less than the minimum price")
//     } else {
//       setTempValues([tempValues[0], newMax])
//     }
//   }

//   return (
//     <div className='flex flex-col items-stretch'>
//       <div className='flex justify-around text-2xl text-bubble-gum font-poppins font-extrabold ml-5 md:ml-10'>
//         <input
//           type='text'
//           value={`$${tempValues[0]}`}
//           onChange={handleMinChange}
//           className='w-24 rounded-md px-2 py-1 outline-none focus:outline-[#FF8A57] focus:outline-offset-2'
//         />
//         <input
//           type='text'
//           value={`$${tempValues[1]}`}
//           onChange={handleMaxChange}
//           className='w-24 rounded-md px-2 py-1 outline-none focus:outline-[#FF8A57] focus:outline-offset-2'
//         />
//       </div>
//       <div className='mt-4 mx-4 md:mx-8 w-[90%]'>
//         <Slider
//           value={tempValues}
//           onChange={handleChange}
//           step={1}
//           min={0}
//           max={1000}
//           aria-labelledby='range-slider'
//           disableRipple
//           classes={{
//             thumb: "bg-[#7874F2] p-3 rounded-full outline-none cursor-pointer",
//             rail: "py-1.5 bg-[#FF8A57] rounded-full",
//             track: "py-1.5 bg-[#FF8A57] rounded-full",
//           }}
//         />
//       </div>
//     </div>
//   )
// }
