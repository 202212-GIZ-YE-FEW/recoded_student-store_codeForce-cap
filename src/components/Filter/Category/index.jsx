import { withTranslation } from "next-i18next"
import { useState } from "react"

import ProductList from "@/components/ProductList"

import { useGeneralListings } from "@/utils/store"

// import products from "@/components/ProductList/products"
import Buttons from "../Buttons"

function CategoryFilter() {
  const { data } = useGeneralListings()
  const filters = [
    {
      filter: "Book",
      image: "/images/Book.png",
      alt: "Book",
    },
    {
      filter: "Furniture",
      image: "/images/Furniture.png",
      alt: "Furniture",
    },
    {
      filter: "Electronics",
      image: "/images/Electronic.png",
      alt: "Electronics",
    },
    {
      filter: "Two-wheeler",
      image: "/images/Two-wheeler.png",
      alt: "Two-wheeler",
    },
  ]

  const [selectedFilter, setSelectedFilter] = useState("")

  const handleCategoryFilter = (filter) => setSelectedFilter(filter)

  return (
    <div>
      <Buttons filters={filters} handleCategoryFilter={handleCategoryFilter} />
      <ProductList selectedFilter={selectedFilter} products={data} />
    </div>
  )
}

export default withTranslation("index")(CategoryFilter)
