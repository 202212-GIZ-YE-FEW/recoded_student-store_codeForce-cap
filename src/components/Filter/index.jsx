import Category from "./Category"
import { SectionWrapper } from "../hoc"

function Filter({ handleCategoryFilter }) {
  return (
    <div>
      <Category handleCategoryFilter={handleCategoryFilter} />
    </div>
  )
}
export default SectionWrapper(Filter)
