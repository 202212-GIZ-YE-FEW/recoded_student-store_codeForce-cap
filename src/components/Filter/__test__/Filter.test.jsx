import renderer from "react-test-renderer"

import Filter from ".."

it("renders correctly", () => {
  const tree = renderer.create(<Filter />).toJSON()
  expect(tree).toMatchSnapshot()
})
