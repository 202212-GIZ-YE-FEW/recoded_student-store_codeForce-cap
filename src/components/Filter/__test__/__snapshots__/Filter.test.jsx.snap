// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <section
    className="max-w-[1250px] mx-auto my-[50px]"
  >
    <div>
      <label>
        Filter by:
      </label>
      <div>
        <button
          onClick={[Function]}
        >
          All
        </button>
        <button
          className="filter-Books"
          onClick={[Function]}
        >
          Books
        </button>
        <button
          className="filter-Furniture"
          onClick={[Function]}
        >
          Furniture
        </button>
        <button
          className="filter-Electronics"
          onClick={[Function]}
        >
          Electronics
        </button>
        <button
          className="filter-Two-wheeler"
          onClick={[Function]}
        >
          Two-wheeler
        </button>
      </div>
      <div>
        <div
          className="carts grid xl:grid-cols-4 lg:grid-cols-4 md:grid-cols-2 sm:grid-cols-1 m-auto w-full"
        >
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Books
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Furniture
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Furniture
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Electronics
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Two-wheeler
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Two-wheeler
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Two-wheeler
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Two-wheeler
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Two-wheeler
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Two-wheeler
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Electronics
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Electronics
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Electronics
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Electronics
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Electronics
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            className="cart mx-3 mb-10 border rounded-lg shadow-lg"
          >
            <img
              alt="Moog Synthesizer"
              className="rounded-t-lg shadow-lg w-full"
              data-nimg="1"
              decoding="async"
              height={211.41}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={258}
            />
            <div
              className="mx-3 text-center"
            >
              <div
                className="info flex justify-between my-4 mx-3"
              >
                <div
                  className="text-left"
                >
                  <h2
                    className="font-semibold"
                  >
                    Moog Synthesizer
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Electronics
                  </p>
                </div>
                <div>
                  <h2
                    className="font-extrabold text-xl"
                  >
                    $
                    468.00
                  </h2>
                  <p
                    className="font-extralight text-xs"
                  >
                    Istanbul / Turkey
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
